Aptoide/aptoide-client-v8
# Feature/mob 401 editorial comments

**What does this PR do?**

   This PR builds the baseline for comments (more specifically in Editorial). It only includes top-level comments (no replies) with endless scrolling and filters fully working. Other functionality is not implemented at this point.
   For this to be implemented in Editorial, a major refactor had to be done. This involved having only one recyclerview in editorial with every component being an item (editorialcontent, reactions, comments). Additionally, the way we presented the data in EditorialPresenter and manager had to be rewritten as well.

   Some other important notes:
   * Initially I wanted to do a specific presenter for comments but here I haven't done that yet as I didn't find a good straightforward solution for that. In interest of time, I haven't done that, and for now every comment event handler is subscribed in EditorialPresenter.
   * Reactions have a specific presenter however I plan to eventually improve it.
   * Recyclerview is implemented with Epoxy.
   * Comments are being memory cached only. This might present an issue as I'm not clearing up this memory throughout the app lifecycle. It will persist until process death. It will probably not cause problems, but if the user sees a lot of comments, it may occupy a relevant amount of memory.
   * There's no comments for Editorial API yet. So for now the comments in "apps" store is being used for testing.
   * A different animation was used for the bottom card in Editorial. 

**Database changed?**

    No

**How should this be manually tested?**

  Go to any editorial and check the comments and if everything works correctly as specified. Additionally check an editorial with only one app and check if the bottom card animation is ok. Everything should work as expected.

**What are the relevant tickets?**

   [MOB-401](https://aptoide.atlassian.net/browse/MOB-401)


**Code Review Checklist**
